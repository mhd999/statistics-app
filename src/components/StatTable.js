// @flow

import React from 'react';
import ReactTable from 'react-table';
import StatTags from './StatTags';
import RowText from './RowText';
import RowLink from './RowLink';
import RowNumber from './RowNumber';

type Props = {
  statistics: stat[],
  loading: boolean,
  classname: string,
  showPagination: boolean,
  pageSize: number,
};

const StatTable = (props: Props) => {
  
  const theadStyle = {
    background: '#d4d4d4',
    cursor: 'default',
  };
  
  const columnsList = 
  [
    {header: 'Title ↓ ↑', accessor: 'title', sortable: true, sort: 'asc', style: {width:400}, render: ({value}) => <RowText text={value} />}, 
    {header: 'Tags', accessor: 'tags', style: {width:400} , render: ({value}) => <StatTags tags={value} />}, 
    {header: 'Id', accessor: 'id',  style: {width:50, textAlign: 'center'} , render: ({value}) => <RowNumber number={value} />}, 
    {header: 'CSV', accessor: 'csvURI', style: {width:200, textAlign: 'center'}, render: ({value}) => <RowLink link={value} text='CSV' />},
    {header: 'JSON', accessor: 'jsonURI', style: {width:200, textAlign: 'center'}, render: ({value}) =><RowLink link={value} text='JSON' />} 
 ];

  return <ReactTable
    showPagination={props.showPagination}
    loading={props.loading}
    data={props.statistics}
    columns={columnsList}
    className={props.classname}
    pageSize={props.pageSize}
    theadStyle={theadStyle}/>
};

export default StatTable;