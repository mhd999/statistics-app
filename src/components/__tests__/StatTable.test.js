import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import StatTable from '../StatTable';

const columnsList = [
  {header: 'key1', accessor: 'label1' },
  {header: 'key2', accessor: 'label2' },
  {header: 'key3', accessor: 'label3' },
  {header: 'key4', accessor: 'label4' },
];

const statissticList = [
  {
    "id":"1104",
    "title":"Befolkningsutvikling. Hele landet, siste kvartal",
    "datasettURI":"http://data.ssb.no/api/v0/dataset/1104/?lang=no",
    "jsonURI":"http://data.ssb.no/api/v0/dataset/1104.json?lang=no",
    "csvURI":"http://data.ssb.no/api/v0/dataset/1104.csv?lang=no",
    "tags":["01222","befolkning","befolkningsvekst","d\u00f8de","flytting","folkemengde","f\u00f8dte","innvandring","utvandring"],
    "statistic":"http://www.ssb.no/folkemengde"
 }
];

const props = {
  showPagination: false,
  loading: false,
  data: statissticList,
  columns: columnsList,
  className: 'classname',
  pageSize: statissticList.length,
};


describe('<StatTable />', () => {
  it('renders a statisstic table', () => {
    const component = shallow(
      <StatTable {...props} />
    );
    const tree = shallowToJson(component);
    expect(tree).toMatchSnapshot();
  });
});
