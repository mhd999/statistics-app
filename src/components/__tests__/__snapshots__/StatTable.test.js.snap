exports[`<StatTable /> renders a statisstic table 1`] = `
<src
  className="-striped -highlight"
  column={
    Object {
      "className": "",
      "headerClassName": "",
      "headerInnerClassName": "",
      "headerInnerStyle": Object {},
      "headerStyle": Object {},
      "innerClassName": "",
      "innerStyle": Object {},
      "show": true,
      "sortable": true,
      "style": Object {},
    }
  }
  columns={
    Array [
      Object {
        "accessor": "title",
        "header": "Title ↓ ↑",
        "render": [Function],
        "sort": "asc",
        "sortable": true,
        "style": Object {
          "width": 400,
        },
      },
      Object {
        "accessor": "tags",
        "header": "Tags",
        "render": [Function],
        "style": Object {
          "width": 400,
        },
      },
      Object {
        "accessor": "id",
        "header": "Id",
        "render": [Function],
        "style": Object {
          "textAlign": "center",
          "width": 50,
        },
      },
      Object {
        "accessor": "csvURI",
        "header": "CSV",
        "render": [Function],
        "style": Object {
          "textAlign": "center",
          "width": 200,
        },
      },
      Object {
        "accessor": "jsonURI",
        "header": "JSON",
        "render": [Function],
        "style": Object {
          "textAlign": "center",
          "width": 200,
        },
      },
    ]
  }
  data={Array []}
  loading={false}
  loadingComponent={[Function]}
  loadingText="Loading..."
  nextComponent={null}
  nextText="Next"
  onChange={[Function]}
  pageSize={1}
  pageSizeOptions={
    Array [
      5,
      10,
      20,
      25,
      50,
      100,
    ]
  }
  paginationClassName=""
  paginationStyle={Object {}}
  previousComponent={null}
  previousText="Previous"
  showPageJump={true}
  showPageSizeOptions={true}
  showPagination={false}
  style={Object {}}
  tableClassName=""
  tableComponent={[Function]}
  tableStyle={Object {}}
  tbodyClassName=""
  tbodyComponent={[Function]}
  tbodyStyle={Object {}}
  tdClassName=""
  tdComponent={[Function]}
  tdStyle={Object {}}
  thClassName=""
  thComponent={[Function]}
  thGroupClassName=""
  thStyle={Object {}}
  theadClassName=""
  theadComponent={[Function]}
  theadStyle={
    Object {
      "background": "#d4d4d4",
      "cursor": "default",
    }
  }
  trClassCallback={[Function]}
  trClassName=""
  trComponent={[Function]}
  trStyle={Object {}}
  trStyleCallback={[Function]} />
`;
